PowerShell Contributing.md

```diff
@@ -0,0 +1,399 @@
+
+To do small contributions like typos, just navigate to that file, click the 'Edit' button, and GitHub will create a copy (fork) on your own GitHub for you.
+When you are satisfied with your changes, you can save them, and submit a pull request to the _staging_ branch of the PowerShell-Docs repository.
+
+
+For bigger contributions, you are better off cloning your fork locally to your computer.
+This enables you to use VSCode and other editing tools.
+A guide on how to do that can be found [here](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/contributing/1-GET-STARTED.md#making-major-edits-to-existing-topics).
+
+With more examples, less typos, clearer wording, PowerShell can be easier to learn and use for everyone!
+
+### 7. Write Pester tests
+
+> Requires
+> 
+> - GitHub account
+> - Sign [Microsoft Contributor License Agreement](https://cla.opensource.microsoft.com/)
+> - Git and some knowledge of it
+> - Knowledge of GitHub workflow (fork > clone > PR)
+> - Know how to run tests
+> - Pester 4.2 or higher
+> - Pester knowledge 
+
+Within the PowerShell project, many parts are tested automatically.
+A big part of that is done with Pester.
+See [here](https://github.com/PowerShell/PowerShell/tree/master/test/powershell).
+
+
+Many CmdLets could use better or more extensive tests to check if they keep working correctly when someone would change them in the future.
+So the Pester tests for CmdLets are kind of a safeguard against regression.
+When there are more/better tests for CmdLets, you can possibly prevent future bugs!
+
+See also "Details about contributing to the PowerShell/PowerShell repo", especially the section on how to run tests.
+
+### 8. Write C# code for PowerShell
+
+> Requires
+> 
+> - GitHub account
+> - Sign [Microsoft Contributor License Agreement](https://cla.opensource.microsoft.com/)
+> - Git and some knowledge of it
+> - Knowledge of GitHub workflow (fork > clone > PR)
+> - Pester knowledge/know how to run test suite
+> - An Integrated Development Environment that supports C#, like VSCode or Visual Studio.
+> - C# knowledge
+
+Well, duh :) When you solve an issue from the PowerShell/PowerShell repo by submitting code, you can make people's lives easier! 
+
+Depending on your level of C# knowledge and experience, pick an issue that suits you, and let others know by commenting on the issue, that you would like to take it on.
+
+
+> I picked a very small [issue](https://github.com/PowerShell/PowerShell/pull/9901), adding a `-Raw` switch to `Select-String`, because I did not know what to expect from the process, and from the code.
+That worked well for me.
+
+
+#### Editor
+
+You can use VSCode with the [C# extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode.csharp), or you can use any version of Visual Studio.
+
+
+#### Find code for a CmdLet
```

Comment:
```suggestion
#### Find code for a Cmdlet
```
There's probably more of these, but I'm about to hit the limit for allowed comments in a single review 😂

Reply:
I'll do another pass this weekend if you like. 🙂

Reply:
Yes please, your feedback is awesome! I love your eye for detail. And your insight in the process is also very valuable. I'm going to finish processing your comments tomorrow morning, and maybe polish some things here and there. If you can find time to have a look after that, that would be great. 
I find I'm juggling between describing things in general vs detailed steps/tips, I will look into that. Thoughts welcome on that too.